src/snowflake/snowpark/lineage.py

```diff
@@ -0,0 +1,509 @@
+                ):
+                    parent_name = properties[_ObjectField.PARENT_NAME_DEPRECATED]
+                else:
+                    raise SnowparkClientExceptionMessages.SERVER_FAILED_FETCH_LINEAGE(
+                        f"missing name/version field for domain {graph_entity[_ObjectField.USER_DOMAIN]}."
+                    )
+                return (f"{db}.{schema}.{parent_name}", name)
+            else:
+                raise SnowparkClientExceptionMessages.SERVER_FAILED_FETCH_LINEAGE(
+                    f"missing name/version field for domain {graph_entity[_ObjectField.USER_DOMAIN]}."
+                )
+
+        return (f"{db}.{schema}.{name}", None)
+
+    def _get_user_entity(self, graph_entity) -> str:
+        """
+        Transforms the given graph entity into a user visible entity.
+        """
+        name, version = self._get_name_and_version(graph_entity)
+
+        domain = (
+            graph_entity.get(_ObjectField.USER_DOMAIN)
+            or graph_entity.get(_ObjectField.REFINED_DOMAIN)
+            or graph_entity.get(_ObjectField.DOMAIN)
+        )
+
+        if _ObjectField.CREATED_ON not in graph_entity:
+            raise SnowparkClientExceptionMessages.SERVER_FAILED_FETCH_LINEAGE(
+                f"missing {_ObjectField.CREATED_ON} property."
+            )
+
+        if _ObjectField.STATUS not in graph_entity:
+            raise SnowparkClientExceptionMessages.SERVER_FAILED_FETCH_LINEAGE(
+                f"missing {_ObjectField.STATUS} property."
+            )
+
+        user_entity = {
+            _ObjectField.NAME: name,
+            _ObjectField.DOMAIN: domain,
+            _ObjectField.CREATED_ON: graph_entity[_ObjectField.CREATED_ON],
+            _ObjectField.STATUS: graph_entity[_ObjectField.STATUS],
+        }
+
+        if version:
+            user_entity[_ObjectField.VERSION] = version
+
+        return user_entity
+
+    def _get_result_dataframe(
+        self, lineage_trace: List[Tuple[VariantType, VariantType, StringType, int]]
+    ) -> "snowflake.snowpark.dataframe.DataFrame":
+        """
+        Constructs a dataframe of lineage results.
+        """
+        transformed_results = []
+
+        for edge in lineage_trace:
+            transformed_results.append(
+                (
+                    self._get_user_entity(edge[0]),
+                    self._get_user_entity(edge[1]),
+                    edge[2].value.capitalize(),
+                    edge[3],
+                )
+            )
+
+        schema = StructType(
+            [
+                StructField("source_object", VariantType()),
+                StructField("target_object", VariantType()),
+                StructField("lineage", StringType()),
+                StructField("depth", IntegerType()),
+            ]
+        )
+        return self._session.create_dataframe(transformed_results, schema=schema)
+
+    @private_preview(version="1.16.0")
+    def trace(
```

Comment:
It's not a must for now but I prefer using a LogicalPlan like other places. Could you help create a followup JIRA?

Reply:
Created a JIRA https://snowflakecomputing.atlassian.net/browse/SNOW-1355760. I have put this in Q2 plan.